### ISSUE TITLE
RollupJS error on file util/isArrayLike.js

### ISSUE THREAD

**Issue by noemi-salaun** [ReactiveX/rxjs#2403]
**RxJS version:**
5.2.0

**Code to reproduce:**
Run RollupJS in a project that uses RxJS.
Use rollup plugin `rollup-plugin-commonjs` with config
```
plugins  : [
        commonjs({
            include     : [
                'node_modules/rxjs/**'
            ]
        }
]
```

**Expected behavior:**
Bundling with RollupJS should work

**Actual behavior:**
```
Error: Unexpected token
node_modules/rxjs/util/isArrayLike.js (2:78)
1: "use strict";
2: var isArrayLike_1 = function (x) { return x && typeof x.length === 'number'; });
                                                                                 ^
3: //# sourceMappingURL=isArrayLike.js.map
```

**Additional information:**
Just removing the parenthesis works for me.

Replace :
```
exports.isArrayLike = (function (x) { return x && typeof x.length === 'number'; });
```
with :
```
exports.isArrayLike = function (x) { return x && typeof x.length === 'number'; };
```

**Comment by kwonoj** [comment]
So this seems like `isArrayLike` is wrapped with parenthesis 
`export const isArrayLike = (<T>(x: any): x is ArrayLike<T> => x && typeof x.length === 'number');`,

but it brings curiousity if it doesn't work with any other codes like this - @noemi-salaun , would you able to try out cjs equivalent such as 
```js
const isArrayLike = ((x) => x && typeof x.length === 'number');
```

and given `rollup-plugin-commonjs` can consume those as well?

**Comment by noemi-salaun** [comment]
I tried with isArrayLike.js looking like :

```js
"use strict";
exports.isArrayLike = ((x) => x && typeof x.length === 'number');
//# sourceMappingURL=isArrayLike.js.map
```

but it's the same error

```
Error: Unexpected token
node_modules/rxjs/util/isArrayLike.js (2:60)
1: "use strict";
2: var isArrayLike_1 = (x) => x && typeof x.length === 'number');
                                                               ^
3: //# sourceMappingURL=isArrayLike.js.map
```

But it seems that `rollup-plugin-commonjs` can handle it.

So I tried without parenthesis :
```js
"use strict";
exports.isArrayLike = (x) => x && typeof x.length === 'number';
//# sourceMappingURL=isArrayLike.js.map
```
It works but `rollup-plugin-uglify` does not like it
```
Error: (uglify plugin) Error transforming bundle with 'uglify' plugin: SyntaxError: Unexpected token: operator (>)
```

**Comment by kwonoj** [comment]
Hm, seems my PR is premature and probably won't work as workaround. 

I'm feeling this is more like rollup side issues, since given module is simple function export and should be able to be consumed. Does rollup confirms it's expected behavior on rollup side? If not, I'd like to suggest check with rollup as well. I assume given error can be reproducible without depends on Rx.

**Comment by noemi-salaun** [comment]
Hm no, I didn't check with rollup people.

I will ask them to see if the issue come from there side.

Thank you for the fast reply :smile:

**Comment by noemi-salaun** [comment]
Hu, someone already take care of it.

https://github.com/rollup/rollup-plugin-commonjs/issues/168

**Comment by kwonoj** [comment]
@noemi-salaun Cool. Let me close this issue for now, feel freely reopen once rollup clarifies if it isn't related with rollup's behavior but need RxJS side code changes.

**Comment by blagerweij** [comment]
This is not a rollup bug, but rxjs. Reverting to version 5.1.1 of rxjs fixes it, since 5.2.0 the problem exists. If you remove the round brackets, it's also fine. Right now this issue is closed, as well as the rollup issue (#168). I suggest this is fixed in rxjs, please re-open the issue and remove the round brackets in isArrayLike function.

**Comment by noemi-salaun** [comment]
There isn't the issue with rxjs 5.1.1 because there isn't the file `util/isArrayLike.js` before version 5.2.0.
The parenthesis aren't the problem, it's the way `rollup-plugin-commonjs` parse it.
